### PR TITLE
StringLengthToEmptyFixer - introduction

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/5005

```
[11:03 AM]-[possum@aquarium]-[~/work/PHP-CS-Fixer]-[git master_StringLengthToEmptyFixer]
$ php php-cs-fixer describe string_length_to_empty
Description of string_length_to_empty rule.
String tests for empty must be done against `''`, not with `strlen`.

Fixer applying this rule is risky.
Risky when `strlen` is overridden, when called using a `stringable` object, also no longer triggers warning when called using non-string(able).

Fixing examples:
```

```diff
--- Original
+++ New
@@ -1,1 +1,1 @@
-<?php $a = 0 === strlen($b) || \STRLEN($c) < 1;
+<?php $a = '' === $b || $c === '';
```

### CODE REVIEW

File: src/Fixer/Alias/RandomApiMigrationFixer.php [hunk 1]
```diff
@@ -81,14 +81,6 @@ public function getDefinition(): FixerDefinitionInterface
         );
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function isCandidate(Tokens $tokens): bool
```
Comment (keradus) [inline src/Fixer/Alias/RandomApiMigrationFixer.php:87]: why dropping ?

File: src/Fixer/CastNotation/ModernizeTypesCastingFixer.php [hunk 2]
```diff
@@ -50,14 +50,6 @@ public function getDefinition(): FixerDefinitionInterface
         );
     }
 
-    /**
-     * {@inheritdoc}
-     */
-    public function isCandidate(Tokens $tokens): bool
```
Comment (keradus) [inline src/Fixer/CastNotation/ModernizeTypesCastingFixer.php:56]: why dropping ?

File: src/Fixer/LanguageConstruct/DirConstantFixer.php [hunk 3]
```diff
@@ -44,7 +44,7 @@ public function getDefinition(): FixerDefinitionInterface
      */
     public function isCandidate(Tokens $tokens): bool
     {
-        return $tokens->isTokenKindFound(T_FILE);
+        return $tokens->isAnyTokenKindsFound([T_STRING, T_FILE]);
```
Comment (keradus) [inline src/Fixer/LanguageConstruct/DirConstantFixer.php:47]: if there is T_STRING but no T_FILE, why to run this fixer?

File: src/Fixer/Alias/RandomApiMigrationFixer.php (hunk 1 above)
Comment (SpacePossum) [reply src/Fixer/Alias/RandomApiMigrationFixer.php:87]: it is in the parent class now

File: src/Fixer/CastNotation/ModernizeTypesCastingFixer.php (hunk 2 above)
Comment (SpacePossum) [reply src/Fixer/CastNotation/ModernizeTypesCastingFixer.php:56]: it is in the parent class now

File: src/Fixer/LanguageConstruct/DirConstantFixer.php (hunk 3 above)
Comment (SpacePossum) [reply src/Fixer/LanguageConstruct/DirConstantFixer.php:47]: should be `all kinds found (string, file)` indeed

**Review by keradus — APPROVED** [review]

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**